src/surface.cpp

```diff
@@ -0,0 +1,1235 @@
+#include "surface.h"
+
+#include <array>
+#include <sstream>
+
+#include "error.h"
+#include "hdf5_interface.h"
+#include "xml_interface.h"
+#include <iostream>
```

Comment:
Group this with other standard library includes. Also, you are missing `#include <cmath>` for `abs` and `sqrt`.

Reply:
How do you find these missing includes.  Do you have a compiler flag that makes them obvious, or are you just operating from memory?

Reply:
Actually using `-Ddebug=on` on my local machine with gcc 6.3 caught it (complained about ambiguous overloads of `abs`). I guess gcc 4.8 on Travis is not smart enough.

Reply:
Ah, I'm working with 5.3.0 so I reckon it's about time to move up a few version numbers.

Reply:
Also, I was justing using std::cout for debugging so I've entirely removed this include now.